motr/m0crate/crate_io.c

```diff
@@ -343,8 +343,9 @@ int cr_namei_create(struct m0_workload_io *cwi,
 		    int                    obj_idx,
 		    int                    op_index)
 {
-	return m0_entity_create(&cwi->cwi_pool_id, &obj->ob_entity,
-			        &cti->cti_ops[free_slot]);
+	return m0_entity_create((m0_fid_is_set(&cwi->cwi_pool_id) && 
+                                m0_fid_is_valid(&cwi->cwi_pool_id)) ? &cwi->cwi_pool_id : NULL,
```

Comment:
This code looks a bit awkward. I'd suggest creating a small static function, for example `check_pool_id()`, which would do the same what this ugly ternary operator does:
```c
static struct m0_fid * check_fid(struct m0_fid *id)
{
    if (m0_fid_is_set(id) && m0_fid_is_valid(id))
        return id;
    else
        return NULL;
}
```

Then use it like this:
```c
return m0_entity_create(check_fid(&cwi->cwi_pool_id),
                        &obj->ob_entity, &cti->cti_ops[free_slot]);
```

Isn't it much more readable?

Note: it's very easy to abuse ternary operators, especially in big expressions. Use them with caution.